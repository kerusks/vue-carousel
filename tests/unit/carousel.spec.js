import { mount, shallowMount } from "@vue/test-utils";
import Carousel from "@/components/carousel";

describe("Carousel", () => {
  describe("Carousel mount and unmount", () => {
    it("should mount", () => {
      const wrapper = mount(Carousel);
      expect(wrapper.vm._isMounted).toBeTruthy();
      expect(wrapper).toMatchSnapshot();
    });

    it("should unmount", () => {
      const wrapper = mount(Carousel);
      const carouselInstance = wrapper.vm;
      carouselInstance.$destroy();
      expect(carouselInstance._isDestroyed).toBeTruthy();
      expect(wrapper).toMatchSnapshot();
    });
  });
});
