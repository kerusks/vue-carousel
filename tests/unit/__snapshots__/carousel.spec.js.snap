// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel Carousel mount and unmount should mount 1`] = `
<div class="wrap">
  <div class="header-text-container">
    <h1>Responsive Carousel Test</h1>
  </div>
  <div class="carousel-wrap">
    <ul class="carousel-container" style="transform: translateX(100%);"></ul>
    <div class="mobile-nav prev" style="display: none;"><img src="../assets/arrow.svg"></div>
    <div class="mobile-nav next" style="display: none;"><img src="../assets/arrow.svg"></div>
  </div>
  <div class="lg-screen-nav-container"><button disabled="disabled">Prev</button> <button disabled="disabled">Next</button></div>
</div>
`;

exports[`Carousel Carousel mount and unmount should unmount 1`] = `
<div class="wrap">
  <div class="header-text-container">
    <h1>Responsive Carousel Test</h1>
  </div>
  <div class="carousel-wrap">
    <ul class="carousel-container" style="transform: translateX(100%);"></ul>
    <div class="mobile-nav prev" style="display: none;"><img src="../assets/arrow.svg"></div>
    <div class="mobile-nav next" style="display: none;"><img src="../assets/arrow.svg"></div>
  </div>
  <div class="lg-screen-nav-container"><button disabled="disabled">Prev</button> <button disabled="disabled">Next</button></div>
</div>
`;
